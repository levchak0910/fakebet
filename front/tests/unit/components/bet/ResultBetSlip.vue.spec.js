import {createWrapper, disableFile, cutFromOptions} from "../../__utils__";
import Component from "@/components/bet/betSlip/types/Result";

disableFile();

describe("bet/ResultBetSlip.vue", () => {

    let wrapper, info, result, betSlipRate;
    let mountWrapper = (options = {}) => {
        let {props} = cutFromOptions(options, ["props"]);
        wrapper = createWrapper(Component, {
            stubs: ["app-bet-slip-info"],
            props: {
                totalCoefficient: 10,
                totalSum: 20,
                outcomeSum: 200,
                ...props,
            },
            remove: {computed: ["bets"]},
            ...options,
        });
        info = wrapper.find(".info");
        result = wrapper.find(".result");
        betSlipRate = wrapper.find(".input > input");
    };

    describe("Testing snapshots", () => {

        it("Component itself", () => {
            mountWrapper();

            expect(wrapper.element).toMatchSnapshot();
        });

        it("Show info", () => {
            mountWrapper({data: {showInfo: true}});

            expect(info.element).toMatchSnapshot();
        });

    });

    describe("Testing triggering methods", () => {

        it("toggleShowInfo", () => {
            mountWrapper({methods: ["toggleShowInfo"]});

            result.trigger("click");

            expect(wrapper.vm.toggleShowInfo).toBeCalled();
        });

    });

    describe("Testing computed properties", () => {

        it("totalCoefficientTmpl", () => {
            mountWrapper({props: {totalCoefficient: 9.23674}});

            expect(wrapper.vm.totalCoefficientTmpl).toBe("9.237");
        });

        it("totalSumTmpl", () => {
            mountWrapper({props: {totalSum: 125.1}});

            expect(wrapper.vm.totalSumTmpl).toBe("125.10");
        });

        describe("betSlipResultText", () => {

            let makeIt = (outcome, result) => {
                it(`betSlipResultText = ${result}`, () => {
                    mountWrapper({props: {outcome}});

                    expect(wrapper.vm.betSlipResultText).toBe(result);
                });
            };

            makeIt("waiting", "sum.potential");
            makeIt("win", "Profit");
            makeIt("lose", "sum.lost");

        });

        it("betSlipResultSum", () => {
            mountWrapper({props: {outcomeSum: 53.122}});

            expect(wrapper.vm.betSlipResultSum).toBe("53.12");
        });

        it("betSlipClass", () => {
            mountWrapper({props: {outcome: "classType"}});

            expect(wrapper.vm.betSlipClass).toBe("bet-slip-classType");
        });

    });

    describe("Testing methods", () => {

        it("toggleShowInfo", () => {
            mountWrapper({data: {showInfo: false}});

            wrapper.vm.toggleShowInfo();

            expect(wrapper.vm.showInfo).toBeTruthy();
        });

    });

    describe("Testing hooks", () => {

        it("created", () => {
            mountWrapper({data: {showInfo: false}});

            wrapper.vm.$root.$emit("changeBetSlipInfoVisibility", "visibility status");

            expect(wrapper.vm.showInfo).toBe("visibility status");
        });

    });

});
