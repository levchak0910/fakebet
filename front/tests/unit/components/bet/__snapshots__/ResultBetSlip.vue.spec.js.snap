// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bet/ResultBetSlip.vue Testing snapshots Component itself 1`] = `
<div
  class="bet-slip bet-slip-undefined"
>
  <div
    class="info"
    style="display: none;"
  >
    <app-bet-slip-info-stub
      linemode="result"
    />
  </div>
  <div
    class="result"
  >
    <p
      class="date"
    >
      Date: 
      <b>
        
      </b>
    </p>
    <p
      class="rate"
    >
      Rate: 
      <b>
        
      </b>
    </p>
    <p
      class="coefficient"
    >
      total.coefficient: 
      <b>
        10.000
      </b>
    </p>
    <p
      class="sum"
    >
      : 
      <b>
        200.00
      </b>
    </p>
  </div>
</div>
`;

exports[`bet/ResultBetSlip.vue Testing snapshots Show info 1`] = `
<div
  class="info"
  style=""
>
  <app-bet-slip-info-stub
    linemode="result"
  />
</div>
`;
